### PR TITLE
Add contribution plots and fix scaling and bugs

### PR DESCRIPTION
**Pull request by macedo22**
Added:
-scatter plots for each channel
-individual channel contributions plot
-summation of channel contributions plot

Fixed:
-if no alpha is specified, the alpha used by LassoCV() is retrieved and used
-changed default threshold
-auxdata only includes non-flat channels before and for lasso model fitting
-number of non-zero coefficients is not just number of channels above threshold
-scaling the timeseries (for overlayed plots) using preprocessing.scale instead of timeseries.detrend()

Output: https://ldas-jobs.ligo-la.caltech.edu/~alexandra.macedo/detchar/LASSO/test/1-14-18/
Command used to generate:
```~/git/gwdetchar/bin/gwdetchar-lasso-correlation 1171432800 1171453380 -f ~/public_html/detchar/chans.txt -a .1 -t .01 -O 2 -o ~/public_html/detchar/LASSO/test/1-14-18 -i L1```

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@macedo22, please be careful when mixing indentation styles. This code should be follow [PEP-0008](https://www.python.org/dev/peps/pep-0008/), meaning all indents are 4-spaces. Can you correct (what I presume are) tab indents?

**Comment by macedo22** [comment]
@duncanmmacleod  Yes I can! I will also keep that in mind for the future. Would you like me to close this pull request and submit a new one, or just add another commit to this with the changes?

**Comment by macedo22** [comment]
@duncanmmacleod Hi Duncan, I made the indentation changes in a follow up commit that is now also reflected in the file changes on this pull request.

**Comment by macedo22** [comment]
Since commit 83d40ac:
1) sample output: https://ldas-jobs.ligo-la.caltech.edu/~alexandra.macedo/detchar/LASSO/test/1-26-18/
2) command used to generate:
` ~/git/gwdetchar/bin/gwdetchar-lasso-correlation 1171432800 1171453380 -f ~/public_html/detchar/chans.txt -a .1 -O 2 -o ~/public_html/detchar/LASSO/test/1-26-18 -J 8 -i L1`

**Comment by duncanmmacleod** [comment]
@macedo22, I fixed the CI issue (at least one) in #76, can you rebase your feature branch against the current `master` and push (`--force`) to re-trigger the CI?

**Comment by macedo22** [comment]
@duncanmmacleod I rebased my forked master and feature, and it looks like it worked. Thanks for the --force tip

File: bin/gwdetchar-lasso-correlation [hunk 1]
```diff
@@ -5,6 +5,11 @@
 # This file is part of the GW DetChar python package.
 #
 # GW DetChar is free software: you can redistribute it and/or modify
+#it under terms of Copyright (C) LIGO Scientific Collaboration (2015-)
+#
+# This file is part of the GW DetChar python package.
+#
+# GW DetChar is free software: you can redistribute it and/or modify
```
Comment (duncanmmacleod) [inline bin/gwdetchar-lasso-correlation]: @macedo22, was this duplication intentional? either way it should be removed...

Comment (macedo22) [reply bin/gwdetchar-lasso-correlation]: no, that must have happened by accident when pasting and I didn't catch it. would you like me to add a new commit?

Comment (macedo22) [reply bin/gwdetchar-lasso-correlation]: actually, I believe a typo in my version caused it to merge strangely when I rebased

Comment (duncanmmacleod) [reply bin/gwdetchar-lasso-correlation]: @macedo22, yes, please correct this with a new commit.

File: bin/gwdetchar-lasso-correlation [hunk 2]
```diff
@@ -285,97 +345,316 @@ def process_channel(input_,):
...
+        scatPlot=gwplot()
+        axes = scatPlot.gca()
+        axes.set_xlabel(chan.replace("_","\_") + " [Channel units]", labelpad=30)
+        axes.set_ylabel("Sensitive range [Mpc]")
+        axes.text(.9,.1,"r = " + str('{0:.2}'.format(pcorr)), verticalalignment='bottom', horizontalalignment='right', transform=axes.transAxes, color="black", size=20, bbox=dict(boxstyle='square', facecolor='white', alpha=.75, edgecolor="black"))
+        scatPlot.add_scatter(ts, rangets, color=rangeColor)
+        scatPlot.add_line(ts, rangeFit, color="black")
+        scatPlot.set_figheight(plotHeight)
+        scatPlot.set_figwidth(plotWidth)
+	scatPlot.tight_layout()
```
Comment (duncanmmacleod) [inline bin/gwdetchar-lasso-correlation]: @macedo22, does this line have tabs? the indentation looks inconsistent?

Comment (macedo22) [reply bin/gwdetchar-lasso-correlation]: Yes, it seems a few tabs were still hidden in there. I will remove them.

File: bin/gwdetchar-lasso-correlation [hunk 3]
```diff
@@ -285,97 +345,316 @@ def process_channel(input_,):
...
+        raise
 
 plt.close()
 
 # -- process channels
 pool = multiprocessing.Pool(nprocplot)
-results = pool.map(process_channel, auxdata.iteritems())
-
+results = pool.map(process_channel, enumerate(auxdata.iteritems()))
+
```
Comment (duncanmmacleod) [inline bin/gwdetchar-lasso-correlation]: @macedo22, can you remove this (and any other) trailing whitespace?

**Comment by duncanmmacleod** [comment]
@macedo22, is this now good to go? I think I have run out of pedantic requests.

**Comment by macedo22** [comment]
@duncanmmacleod As far as I can tell, yes! I just double-checked to make sure nothing was accidentally messed with and made a new output page (https://ldas-jobs.ligo-la.caltech.edu/~alexandra.macedo/detchar/LASSO/test/2-1-18/) and it looks as expected.

**Pull request merged**